CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+> - Some cloud providers offer a free tier for the first **100GB** of traffic which is not taken into consideration here for simplicity.
+
+| Cloud Provider | Inbound Traffic | Outbound Traffic |
+|--------|------|------|
+| AWS | 0 $/GB | 0.09 $/GB |
+| GCP | 0 $/GB | 0.11 $/GB |
+| Azure | 0 $/GB | 0.09 $/GB |
+| Average | 0 $/GB | 0.1 $/GB |
+
+
+#### Mithril message diffusion extra networking cost
+
+For a total of `3,000` SPOs sending messages, the extra networking cost incurred for a Cardano full node is:
+
+| Send period | Lower bound | Upper bound |
+| ----------- | ----------- | ----------- |
+| 1 min       | 15 $/month  | 36 $/month  |
+| 2 min       | 8 $/month   | 18 $/month  |
+| 5 min       | 3 $/month   | 8 $/month   |
+| 10 min      | 2 $/month   | 4 $/month   |
+
+### Protocol authentication
+
+#### Message authentication mechanism
+The message body is signed with the KES key of the SPO. This signature and the operational certificate of the SPO are appended to the message which is diffused.
+
+Before being diffused to other peers, an incoming message must be verified by the receiving node. This is done with the following steps:
+- Verify that the operational certificate is valid by checking that the KES verification key is signed by cold secret key.
+- Verify the KES signature of the message body with the KES verification key from the operational certificate.
+- Compute the SPO pool id by hashing the cold verification key from the operational certificate.
+- Make sure that this pool id is part of the stake distribution (the network layer will need to have access to this information).
+
+If any of these step fails, the message is considered as invalid, which is a protocol violation.
+
+> [!WARNING]  
+> We also probably need to make sure that the KES key used to sign is from the latest rotation:
+> - either the last seen opcert number in the block headers of the chain.
+> - or the last seen opcert number from a previous message diffused.
+> 
+> If the opcert number received is strictly lower than the previous one which has been seen, it > should be considered as a protocol violation.
+
+#### Cost of authentication
+
+> [!NOTE]
+> Computations are based on a **2 ms** verification delay per message on a CPU core.
+
+For a total of **3,100** Cardano SPOs, there will be **1,550** messages sent per round of signature: 
+
+- the number of messages received by a node which need to be verified is:
+
+| Send period | Messages sent |
+| ----------- | ------------- |
+| 1 min       | 64 M/month    |
+| 2 min       | 32 M/month    |
+| 5 min       | 13 M/month    |
+| 10 min      | 7 M/month     |
+
+- the extra CPU time for verification for a Cardano full node is:
+
+| Send period | CPU core usage |
+| ----------- | -------------- |
+| 1 min       | 5%             |
+| 2 min       | 2.5%           |
+| 5 min       | 1.0%           |
+| 10 min      | 0.5%           |
```

Comment:
Where are these numbers coming from? Also the 2ms cpu time per messages seems quite arbitrary?

Breaking this down to the individual task of authentication is a bit weird IMO. It's good that we mention that verifying and forwarding messages is not free, but this does not seem to be the full picture still.

Reply:
The numbers are the computed based on the number of messages received per month and the time needed to verify the KES signature for each message. I updated the text to make it more clear how the computations are made. 

The `2ms` CPU is the average KES signature verification time IIRC (it can be adjusted if this is not the correct value).